### PR TITLE
feat(issueVariant): allow for creation of issue variant with only a severity rating

### LINKED ISSUE #361: feat(issueVariant): Allow for issue variant creation with just severity rating
**Task Description**
Currently the severity input type for creating issue variants only accepts a cvss vector. Add a rating field alongside vector and update the resolver methods to allow for creating a new issue variant where vector is not provided. 

**Acceptance Criteria:**

- [x] Add a "rating" field of type SeverityValues to SeverityInput
- [x] Add a new "NewSeverityFromRating" function that creates a severity with only the rating field and not the vector
- [x] Add logic to handle between when a vector is provided and when only a rating is provided
- [x] Be able to create a new IssueVariant with only a rating

### PR DESCRIPTION
**Pull request by BlakePatterson**
## Description
As part of the Openstack scanner development, we needed to be able to create issue variants with a severity that just has a rating and no cvss vector. 

This PR adds a "rating" field to SeverityInput of type SeverityValues and also adds logic throughout the resolver implementation to handle the separate cases of whether a vector has been passed in SeverityInput or just a rating. There is also a new query in the queryCollection/issueVariant folder for the case of creating a query with only a severity rating and no vector. 

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

- Closes #361 (issue)

## Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help
- [ ] Separate ticket for tests # (issue/pr)

I created a new query for the case of creating an issue variant with only a severity rating and added a new e2e test case for it. I also updated any tests involving queries that contained SeverityInput wherever necessary. 

All tests passed locally, and I also tested various queries in the playground to ensure issue variant creation worked. 

## Added to documentation?

- [ ] 📜 README.md
- [ ] 🤝 Documentation pages updated
- [x] 🙅 no documentation needed
- [ ] (if applicable) generated OpenAPI docs for CRD changes

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: internal/entity/common.go [hunk 1]
```diff
@@ -150,6 +150,26 @@ type Cursor struct {
 	Limit     int
 }
 
+func NewSeverityFromRating(rating SeverityValues) Severity {
```
Comment (MR2011) [inline internal/entity/common.go:153]: Please add a comment where this values come from. There might also be an adjustment depending on the CVSS version.
https://nvd.nist.gov/vuln-metrics/cvss

File: internal/e2e/evidence_query_test.go [hunk 2]
```diff
@@ -317,8 +317,12 @@ var _ = Describe("Updating evidence via API", Label("e2e", "Evidences"), func()
 				evidence.Description = "New Description"
 
 				req.Var("id", fmt.Sprintf("%d", evidence.Id))
-				req.Var("input", map[string]string{
+				req.Var("input", map[string]interface{}{
 					"description": evidence.Description,
+					"severity": model.SeverityInput{
```
Comment (MR2011) [inline internal/e2e/evidence_query_test.go:322]: Why is this needed in the evidence test?

File: internal/e2e/issue_variant_query_test.go [hunk 3]
```diff
@@ -312,8 +349,12 @@ var _ = Describe("Updating issueVariant via API", Label("e2e", "IssueVariants"),
 				issueVariant.SecondaryName = "SecretIssueVariant"
 
 				req.Var("id", fmt.Sprintf("%d", issueVariant.Id))
-				req.Var("input", map[string]string{
+				req.Var("input", map[string]interface{}{
 					"secondaryName": issueVariant.SecondaryName,
+					"severity": model.SeverityInput{
+						Rating: &model.AllSeverityValues[0],
+						Vector: &issueVariant.Severity.Cvss.Vector,
```
Comment (MR2011) [inline internal/e2e/issue_variant_query_test.go:356]: Do we check if rating and vector match?

Comment (BlakePatterson) [reply internal/e2e/issue_variant_query_test.go:356]: I added logic now to check if rating & vector don't match when they are both passed. If they match it works like normal, if not the issue variant creation errors out.

File: internal/entity/common.go (hunk 1 above)
Comment (BlakePatterson) [reply internal/entity/common.go:153]: I added a comment about the CVSS specification with links, but I'm not sure how I would handle doing adjustments based on which CVSS specification since there is no information being passed other than the rating. Should I just do something else here instead?

File: internal/e2e/evidence_query_test.go (hunk 2 above)
Comment (BlakePatterson) [reply internal/e2e/evidence_query_test.go]: I removed this now as it's not needed and works without the severity input now.

File: internal/api/graphql/graph/model/models.go [hunk 4]
```diff
@@ -145,9 +145,24 @@ func NewSeverity(sev entity.Severity) *Severity {
 }
 
 func NewSeverityEntity(severity *SeverityInput) entity.Severity {
-	if severity == nil || severity.Vector == nil {
+	if severity == nil || (severity.Rating == nil && severity.Vector == nil) {
+		// no severity information was passed
 		return entity.Severity{}
 	}
+	if severity.Rating != nil && severity.Vector != nil {
```
Comment (drochow) [inline internal/api/graphql/graph/model/models.go:152]: If a vector is there, let's take the vector by default and simply overrule the "normal" severity. 

@MR2011  @dorneanu  WDYT?

**Review by drochow — COMMENTED** [review]
LGTM, but i would suggest the above 👍

File: internal/api/graphql/graph/model/models.go (hunk 4 above)
Comment (MR2011) [reply internal/api/graphql/graph/model/models.go:152]: I agree

File: internal/entity/common.go (hunk 1 above)
Comment (MR2011) [reply internal/entity/common.go:153]: For now that's okay

File: internal/api/graphql/graph/model/models.go (hunk 4 above)
Comment (BlakePatterson) [reply internal/api/graphql/graph/model/models.go]: I've updated it now so that it just uses the vector and ignores the rating when both are passed.

File: internal/e2e/issue_variant_query_test.go [hunk 5]
```diff
@@ -261,6 +261,43 @@ var _ = Describe("Creating IssueVariant via API", Label("e2e", "IssueVariants"),
...
+				}
+				if err := util2.RequestWithBackoff(func() error { return client.Run(ctx, req, &respData) }); err != nil {
+					logrus.WithError(err).WithField("request", req).Fatalln("Error while unmarshaling")
+				}
+
+				Expect(*respData.IssueVariant.SecondaryName).To(Equal(issueVariant.SecondaryName))
+				Expect(*respData.IssueVariant.Description).To(Equal(issueVariant.Description))
+				Expect(*respData.IssueVariant.IssueRepositoryID).To(Equal(fmt.Sprintf("%d", issueVariant.IssueRepositoryId)))
+				Expect(*respData.IssueVariant.IssueID).To(Equal(fmt.Sprintf("%d", issueVariant.IssueId)))
+				Expect(string(*respData.IssueVariant.Severity.Value)).To(Equal(issueVariant.Severity.Value))
```
Comment (MR2011) [inline internal/e2e/issue_variant_query_test.go:299]: We should check that the Vector is empty after updating the rating

File: internal/e2e/issue_variant_query_test.go [hunk 6]
```diff
@@ -329,6 +366,44 @@ var _ = Describe("Updating issueVariant via API", Label("e2e", "IssueVariants"),
...
+
+				var respData struct {
+					IssueVariant model.IssueVariant `json:"updateIssueVariant"`
+				}
+
+				if err := util2.RequestWithBackoff(func() error { return client.Run(ctx, req, &respData) }); err != nil {
+					logrus.WithError(err).WithField("request", req).Fatalln("Error while unmarshaling")
+				}
+
+				Expect(string(*respData.IssueVariant.Severity.Value)).To(Equal(issueVariant.Severity.Value))
```
Comment (MR2011) [inline internal/e2e/issue_variant_query_test.go:405]: We should check if the vector is empty after updating the rating

**Review by MR2011 — APPROVED** [review]
